Setup Eslint

Roman-Likhomanov opened the pull request:
Setup Eslint

vasiliev-alexey commented on .eslintrc.js:
```diff
@@ -0,0 +1,21 @@
+    'max-len': ['error', {
+      ignoreComments: true,
+    }],
+    'import/prefer-default-export': 'off',
```
Поскольку в коде используется вывод в консоль - то  для уеньшения варнингов помоттреть на     "no-console": "off",

vasiliev-alexey commented on package.json:
```diff
@@ -4,7 +4,9 @@
   "scripts": {
-    "test": "jest"
+    "test": "jest",
+    "lint": "eslint .",
```
прогон   выводит ✖ 43 problems (30 errors) - надо бы пофиксить

vasiliev-alexey commented on src/7/task7.1.test.js:
```diff
@@ -1,33 +1,26 @@
+      ['I love IT!'],
+      ['I love world!'],
+    ];
+    expect([...el.querySelectorAll('ul.history li')].map((el) => el.innerHTML)).toEqual(values.map(([value]) => value));
```
идет сокрытие переменной

vasiliev-alexey commented on src/7/task7.2.test.js:
```diff
@@ -1,31 +1,24 @@
-      });
-    });
+    const button = el.querySelector('button');
+    el.querySelector('input').keydown;
```
Почитать про dispatch event

vasiliev-alexey commented on src/task2.js:
```diff
@@ -1,52 +1,52 @@
+  (b > a) ? console.log(b) : console.log(a);
+}
 
-export function month() { 
-    const month = window.prompt('Enter number');
```
имя переменной совпадает с имением функции - это  скверно

vasiliev-alexey commented on src/task9.js:
```diff
@@ -9,38 +9,35 @@ export function rightTriangle(a, b, c) {
 
   if (a2 + b2 == c2) {
```
прочитать про разницу == и ===

vasiliev-alexey commented on src/task4.js:
```diff
@@ -1,15 +1,15 @@
 export function addage(user) {
```
кемелКейс - это  не надо пониметь - это надо запомнить ;)

vasiliev-alexey commented on src/task5.js:
```diff
@@ -1,32 +1,32 @@
+    const modified = el * 2;
     return modified;
-    });
-    return newArr;
```
можно  обойтись и без это переменной -

vasiliev-alexey commented on src/task5.js:
```diff
@@ -1,32 +1,32 @@
     }
-    
-    let min = arr[0];
-    for (let i = 0; i <= arr.length; i++) {
```
мы 2 раза перебираем массив - может можно сделать за 1 раз ?

vasiliev-alexey commented on src/task6.js:
```diff
@@ -1,25 +1,23 @@
+  return true;
+}
 
 export function pow(a, x) {
```
прочиать на досуге про ** https://learn.javascript.ru/operators

vasiliev-alexey commented on src/task8.js:
```diff
@@ -4,33 +4,32 @@ export function dayOfWeek() {
   const date1 = new Date(`${year1}-${month1}-${day1}`);
-  
+
   // const input2 = '10.05.1999';
```
комменты  - тут не разьяснют логику - почистить

Roman-Likhomanov commented:
Алексей, спасибо! Как раз сделал  "no-console": "off" , просто не
перепушил изменения.

пн, 23 авг. 2021 г. в 22:16, vasiliev-alexey ***@***.***>:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In .eslintrc.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694241805>
> :
>
> > +    es2021: true,
>
> +    "jest/globals": true,
>
> +  },
>
> +  extends: [
>
> +    'airbnb-base',
>
> +  ],
>
> +  parserOptions: {
>
> +    ecmaVersion: 12,
>
> +    sourceType: 'module',
>
> +  },
>
> +  plugins: ["jest"],
>
> +  rules: {
>
> +    'max-len': ['error', {
>
> +      ignoreComments: true,
>
> +    }],
>
> +    'import/prefer-default-export': 'off',
>
>
> Поскольку в коде используется вывод в консоль - то для уеньшения варнингов
> помоттреть на "no-console": "off",
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#pullrequestreview-736480293>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATPVCJUMC3FKYXR6I4YFAQDT6KNCFANCNFSM5CP5D5UA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

vasiliev-alexey commented on src/task9.test.js:
```diff
@@ -1,51 +1,47 @@
+  it('equation with 2 roots', () => {
+    jest
+      .spyOn(window, 'prompt')
+      .mockImplementationOnce(() => '1')
+      .mockImplementationOnce(() => '-8')
+      .mockImplementationOnce(() => '12');
```
можно исключить дублирование лоигки тестов чрез параметризацияю в табличгых тестах 

https://jestjs.io/docs/api#testconcurrenteachtablename-fn-timeout

vasiliev-alexey commented on src/7/task7.1.test.js:
```diff
@@ -1,33 +1,26 @@
-
-     
+    const values = [
+      ['I love JavaScript!'],
+      ['I love IT!'],
+      ['I love world!'],
```
👍

vasiliev-alexey commented on src/task2.test.js:
```diff
@@ -1,27 +1,24 @@
-import {square} from './task2';
-describe("square", () => {
-  it("a circle in a square", () => {
+describe('square', () => {
```
мало тестов - хорошо беы покрыть и корнер кейсы

vasiliev-alexey commented on src/task6.test.js:
```diff
@@ -1,21 +1,18 @@
+describe('isWord', () => {
+  it('number of words', () => {
+    jest.spyOn(window, 'prompt').mockReturnValue('hello John');
     expect(isWord()).toBe(false);
```
а где тест на инверсию ?

vasiliev-alexey commented on src/task6.test.js:
```diff
@@ -1,21 +1,18 @@
-  it("a to the power of x", () => {
+describe('pow', () => {
+  it('a to the power of x', () => {
     expect(pow(2, 10)).toBe(1024);
```
больше тестов

vasiliev-alexey commented on src/task8.test.js:
```diff
@@ -1,25 +1,22 @@
-    const mockDate = new Date("Mon Aug 16 2021 20:40:15 GMT+0300 (Москва, стандартное время)");
+describe('numberOfMinutes', () => {
+  it('numberOfMinutes', () => {
+    const mockDate = new Date('Mon Aug 16 2021 20:40:15 GMT+0300 (Москва, стандартное время)');
     jest
```
а если будет  не российская локаль будет работать ?. может как-о подумать о дргих вариантах, менее локале  зависимых

vasiliev-alexey commented on src/task8.test.js:
```diff
@@ -1,25 +1,22 @@
-  it("young user", () => {
+describe('youngUser', () => {
+  it('young user', () => {
     expect(youngUser('01.06.1990', '10.05.1999')).toBe('10.05.1999');
   });
```
тесты на инверсию . на равернство ?

Roman-Likhomanov commented:
Алексей, спасибо за подсказки. А вы как то относитесь к Otus?

пн, 23 авг. 2021 г. в 22:53, vasiliev-alexey ***@***.***>:

> ***@***.**** approved this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694242790>
> :
>
> > @@ -4,7 +4,9 @@
>
>    "description": "",
>
>    "main": "index.js",
>
>    "scripts": {
>
> -    "test": "jest"
>
> +    "test": "jest",
>
> +    "lint": "eslint .",
>
>
> прогон выводит ✖ 43 problems (30 errors) - надо бы пофиксить
> ------------------------------
>
> In src/7/task7.1.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694243419>
> :
>
> > -        ["I love JavaScript!"],
>
> -        ["I love IT!"],
>
> -        ["I love world!"]
>
> -      ];
>
> -      expect([...el.querySelectorAll("ul.history li")].map((el) => el.innerHTML)
>
> -      ).toEqual(values.map(([value]) => value)
>
> -      );
>
> -    });
>
> -
>
> -
>
> +    const values = [
>
> +      ['I love JavaScript!'],
>
> +      ['I love IT!'],
>
> +      ['I love world!'],
>
> +    ];
>
> +    expect([...el.querySelectorAll('ul.history li')].map((el) => el.innerHTML)).toEqual(values.map(([value]) => value));
>
>
> идет сокрытие переменной
> ------------------------------
>
> In src/7/task7.2.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694244187>
> :
>
> > -  import {showButton} from './task7.2';
>
> -  describe("showButton", () => {
>
> -
>
> -      it("show the button when entering", () => {
>
> -
>
> -      const el = document.createElement('div');
>
> -      showButton(el);
>
> -
>
> -      const button = el.querySelector("button");
>
> -      el.querySelector("input").keydown;
>
> -      expect(button.hidden).toBe(false);
>
> -
>
> -      });
>
> -    });
>
> +    const button = el.querySelector('button');
>
> +    el.querySelector('input').keydown;
>
>
> Почитать про dispatch event
> ------------------------------
>
> In src/task2.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694246560>
> :
>
> > -export function month() {
>
> -    const month = window.prompt('Enter number');
>
>
> имя переменной совпадает с имением функции - это скверно
> ------------------------------
>
> In src/task9.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694247860>
> :
>
> > @@ -9,38 +9,35 @@ export function rightTriangle(a, b, c) {
>
>
>
>    if (a2 + b2 == c2) {
>
>
> прочитать про разницу == и ===
> ------------------------------
>
> In src/task4.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694248623>
> :
>
> > @@ -1,15 +1,15 @@
>
>  export function addage(user) {
>
>
> кемелКейс - это не надо пониметь - это надо запомнить ;)
> ------------------------------
>
> In src/task5.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694250031>
> :
>
> >      return modified;
>
> -    });
>
> -    return newArr;
>
>
> можно обойтись и без это переменной -
> ------------------------------
>
> In src/task5.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694250636>
> :
>
> >      }
>
> -
>
> -    let min = arr[0];
>
> -    for (let i = 0; i <= arr.length; i++) {
>
>
> мы 2 раза перебираем массив - может можно сделать за 1 раз ?
> ------------------------------
>
> In src/task6.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694251644>
> :
>
> >
>
>  export function isWord() {
>
> -    let word = prompt('enter text');
>
> -    let arr = word.split(' ');
>
> -    if (arr.length > 1) {
>
> -        return false;
>
> -    } else {
>
> -        return true;
>
> -    }
>
> -}
>
> +  const word = prompt('enter text');
>
> +  const arr = word.split(' ');
>
> +  if (arr.length > 1) {
>
> +    return false;
>
> +  }
>
> +  return true;
>
> +}
>
>
>
>  export function pow(a, x) {
>
>
> прочиать на досуге про ** https://learn.javascript.ru/operators
> ------------------------------
>
> In src/task8.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694252297>
> :
>
> >
>
>  export function youngUser(input1, input2) {
>
>    // const input1 = '01.06.1990';
>
>    const [day1, month1, year1] = input1.split('.');
>
>    const date1 = new Date(`${year1}-${month1}-${day1}`);
>
> -
>
> +
>
>    // const input2 = '10.05.1999';
>
>
> комменты - тут не разьяснют логику - почистить
> ------------------------------
>
> In src/task9.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694255795>
> :
>
> > +      .mockImplementationOnce(() => '1')
>
> +      .mockImplementationOnce(() => '-8')
>
> +      .mockImplementationOnce(() => '12');
>
>
> можно исключить дублирование лоигки тестов чрез параметризацияю в
> табличгых тестах
>
> https://jestjs.io/docs/api#testconcurrenteachtablename-fn-timeout
> ------------------------------
>
> In src/7/task7.1.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694257159>
> :
>
> > +      ['I love JavaScript!'],
>
> +      ['I love IT!'],
>
> +      ['I love world!'],
>
>
> 👍
> ------------------------------
>
> In src/task2.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694258134>
> :
>
> >      console.log = jest.fn();
>
>      month();
>
>      expect(console.log).toHaveBeenCalledWith('dec');
>
>    });
>
>  });
>
> -
>
> -import {square} from './task2';
>
> -describe("square", () => {
>
> -  it("a circle in a square", () => {
>
> +describe('square', () => {
>
>
> мало тестов - хорошо беы покрыть и корнер кейсы
> ------------------------------
>
> In src/task6.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694258807>
> :
>
> >      expect(diff(23, 5)).toBe(18);
>
>    });
>
>  });
>
> -
>
> -import {isWord} from './task6';
>
> -describe("isWord", () => {
>
> -  it("number of words", () => {
>
> -    jest.spyOn(window, "prompt").mockReturnValue("hello John");
>
> +describe('isWord', () => {
>
> +  it('number of words', () => {
>
> +    jest.spyOn(window, 'prompt').mockReturnValue('hello John');
>
>      expect(isWord()).toBe(false);
>
>
> а где тест на инверсию ?
> ------------------------------
>
> In src/task6.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694259054>
> :
>
> >      expect(isWord()).toBe(false);
>
>    });
>
>  });
>
> -
>
> -import {pow} from './task6';
>
> -describe("pow", () => {
>
> -  it("a to the power of x", () => {
>
> +describe('pow', () => {
>
> +  it('a to the power of x', () => {
>
>      expect(pow(2, 10)).toBe(1024);
>
>
> больше тестов
> ------------------------------
>
> In src/task8.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694259995>
> :
>
> > +    const mockDate = new Date('Mon Aug 16 2021 20:40:15 GMT+0300 (Москва, стандартное время)');
>
>      jest
>
>
> а если будет не российская локаль будет работать ?. может как-о подумать о
> дргих вариантах, менее локале зависимых
> ------------------------------
>
> In src/task8.test.js
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#discussion_r694260251>
> :
>
> >      expect(youngUser('01.06.1990', '10.05.1999')).toBe('10.05.1999');
>
>    });
>
>
> тесты на инверсию . на равернство ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#pullrequestreview-736481385>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATPVCJQBPH56FU3DF2H5EQ3T6KRMZANCNFSM5CP5D5UA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

vasiliev-alexey commented:
> Алексей, спасибо за подсказки. А вы как то относитесь к Otus?

Я обучающийся этого же курса только запуска декабря  2020. Думаю, если  вы дойдете до 6 месяца обучения и у вас в ДЗ будут задания по ревью кода ДЗ, курсов, которые только старатнули   обучение.

Roman-Likhomanov commented:
Круто, и как успехи. Вы в конце июня получается курс закончили? Как
трудоустройство?

вт, 24 авг. 2021 г. в 19:36, vasiliev-alexey ***@***.***>:

> Алексей, спасибо за подсказки. А вы как то относитесь к Otus?
>
> Я обучающийся этого же курса только запуска декабря 2020. Думаю, если вы
> дойдете до 6 месяца обучения и у вас в ДЗ будут задания по ревью кода ДЗ,
> курсов, которые только старатнули обучение.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#issuecomment-904798524>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATPVCJQJ6DZFY6HQMYIB52LT6PDA7ANCNFSM5CP5D5UA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

vasiliev-alexey commented:
> Круто, и как успехи. Вы в конце июня получается курс закончили? Как трудоустройство? 
Нет - наш  запуск на финишной прямой - еще 3 лекции и проект.
Менять работу я не плнировал, для меня это больше по фану.

Совет на будущее - не затягивай с ДЗ, впереди они будут обьемными , но интересными.
Сделать за раз на 100% не получится,  лучше сдлеть базу и сдать ее, а после  менторской сесси исравишь - быстрая обратная свзяь она лучше стимулирует.

Roman-Likhomanov commented:
Понял, ну я не специально затянул. Просто тяжело пока дается, я с 0 начал.
Ну вот я и сдал её с ошибками, сейчас правлю. И да, нужно за следующую
домашку браться.
Спасибо за совет. Я тоже так подумал, что лучше сдавать и докручивать.
А у вас курс что ли 8 мес идет?

вт, 24 авг. 2021 г., 20:59 vasiliev-alexey ***@***.***>:

> Круто, и как успехи. Вы в конце июня получается курс закончили? Как
> трудоустройство?
> Нет - наш запуск на финишной прямой - еще 3 лекции и проект.
> Менять работу я не плнировал, для меня это больше по фану.
>
> Совет на будущее - не затягивай с ДЗ, впереди они будут обьемными , но
> интересными.
> Сделать за раз на 100% не получится, лучше сдлеть базу и сдать ее, а после
> менторской сесси исравишь - быстрая обратная свзяь она лучше стимулирует.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#issuecomment-904856553>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATPVCJQGE3GDMHZ2KY5G3CDT6PMXFANCNFSM5CP5D5UA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

vasiliev-alexey commented:
> А у вас курс что ли 8 мес идет?

Да, до пересмотра программы  , курс был длиной 9 месяцев

Roman-Likhomanov commented:
Теперь стало понятно) Ну у меня тоже 9 месяцев будет. Я в марте пошел, но
возникли проблемы, Василий сказал нужно задачи нарешивать в кодевар. И я по
итогу отстал. В итоге сейчас в этот поток перешел.

вт, 24 авг. 2021 г. в 21:09, vasiliev-alexey ***@***.***>:

> А у вас курс что ли 8 мес идет?
>
> Да, до пересмотра программы , курс был длиной 9 месяцев
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Roman-Likhomanov/task-lesson-7/pull/2#issuecomment-904863327>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATPVCJVVNU5SGPJGXODEFQDT6PN5RANCNFSM5CP5D5UA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>
